### PR TITLE
FIX: Remove last_unread_post excerpt logic for bookmarks

### PR DESCRIPTION
**Pull request by martin-brennan**
The logic to determine what post excerpt to show for
a topic-level bookmark based on the last unread post
was complex and slow, so we decided to remove it and
always just use the first post excerpt.

This commit also fixes an issue where a couple of
instances of for_topic were missed when doing the
Bookmarkable refactors, so:

1. Clicking the topic bookmark link was not taking
   the user to the last unread post
2. When replying to a topic where there was a topic
   level bookmark with the auto delete preference
   of "on owner reply", we were not removing the
   bookmark from the UI correctly.

A test has been added for the former, the latter would
be quite time-consuming to test and not really worth
it considering it's quite an edge case UI bug.

### CODE REVIEW

**Review by ZogStriP — APPROVED** [review]
🔥

File: app/serializers/user_topic_bookmark_serializer.rb [hunk 1]
```diff
@@ -28,28 +27,7 @@ def raw
...
-    first_unread_post = sorted_regular_posts.find do |post|
-      post.post_number >= post_number
-    end
-
-    # if first_unread_cooked is blank this likely means that the last
-    # read post was either deleted or is a small action post.
-    # in this case we should just get the last regular post and
-    # use that for the cooked value so we have something to show
-    (first_unread_post || sorted_regular_posts.last).cooked
+    first_post.cooked
```
Comment (davidtaylorhq) [inline app/serializers/user_topic_bookmark_serializer.rb:30]: I think we need to switch from preloading `posts`, to preloading the `first_post` relation here:

https://github.com/discourse/discourse/blob/72ea00d0cc69e49017ecb363f2dc1a39aa2dea14/app/services/topic_bookmarkable.rb#L14-L16

So that would become:

```ruby
def self.preload_associations
  [:tags, { first_post: :user }]
end
```

Comment (martin-brennan) [reply app/serializers/user_topic_bookmark_serializer.rb:30]: Ah good catch! Thanks, will change :+1:
